### PR TITLE
CPO-1785 Added equals, hashcode, != and == overloads for Address.

### PR DESCRIPTION
**Pull request by cstait**

### CODE REVIEW

File: CertiPay.Payroll.Common/Address.cs [hunk 1]
```diff
@@ -53,5 +53,49 @@ public Address()
...
+            {
+                return true;
+            }
+            if ((object)Left == null || (object)Right == null)
+            {
+                return false;
+            }
+
+            return (
+                Left.Address1 == Right.Address1 &&
```
Comment (mattgwagner) [inline CertiPay.Payroll.Common/Address.cs]: Should probably consider case-insensitivity, or at least document that assumption.

File: CertiPay.Payroll.Common/Address.cs [hunk 2]
```diff
@@ -53,5 +53,49 @@ public Address()
...
+            if ((Object)objAddress == null)
+            {
+                return false;
+            }
+            return (objAddress == this);
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```
Comment (mattgwagner) [inline CertiPay.Payroll.Common/Address.cs]: Either do it semi-legitimately or don't implement it, this doesn't really provide anything of use.

Comment (cstait) [inline CertiPay.Payroll.Common/Address.cs]: I think I'll opt to remove it for now, there isn't really any immutable properties on this object to base a hash code off of.

**Comment by mattgwagner** [comment]
@cstait looks good -- before I merge it, can you throw a unit test or two in there to verify that it behaves the way we expect? Both passing case and failing case, please.

**Pull request merged**